client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx

```diff
@@ -1,40 +1,106 @@
+import { useTranslate } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import theme from 'calypso/my-sites/plugins/marketplace/theme';
 import Masterbar from 'calypso/layout/masterbar/masterbar';
-import { getStatusForPlugin } from 'calypso/state/plugins/installed/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import QueryJetpackPlugins from 'calypso/components/data/query-jetpack-plugins';
-
+import { initiateThemeTransfer } from 'calypso/state/themes/actions';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
 /**
  * Style dependencies
  */
 import 'calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status/style.scss';
+import { getPurchaseFlowState } from 'calypso/state/plugins/marketplace/selectors';
 
 const StyledProgressBar = styled( ProgressBar )`
 	margin: 20px 0px;
 `;
 
-function WrappedMarketplacePluginSetup( {
-	pluginSlug = 'wordpress-seo',
-}: {
-	pluginSlug?: string;
-} ): JSX.Element {
+// Stages
+
+function WrappedMarketplacePluginSetup(): JSX.Element {
+	const translate = useTranslate();
+
+	const STAGE_1 = translate( 'Installing plugin' );
+	const STAGE_2 = translate( 'Activating plugin' );
+	const stages = [ STAGE_1, STAGE_2 ];
+
+	const [ currentStage, setCurrentStage ] = useState( STAGE_1 );
+	const [ simulatedProgressPercentage, setSimulatedProgressPercentage ] = useState( 0 );
+
+	const interval = 2000;
+
+	const dispatch = useDispatch();
 	const selectedSite = useSelector( getSelectedSite );
+	const transferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, selectedSite?.ID )
+	);
+	const { pluginSlugToBeInstalled, isPluginInstalledDuringPurchase } = useSelector(
+		getPurchaseFlowState
+	);
+
+	useEffect( () => {
+		if ( pluginSlugToBeInstalled ) {
+			dispatch( initiateThemeTransfer( selectedSite.ID, null, pluginSlugToBeInstalled ) );
+		} else if ( isPluginInstalledDuringPurchase ) {
+			alert( 'To be implemented: polling wait to be implemented to check for transfer' );
+		} else {
+			// Invalid State redirect to Yoast marketplace page for now, and maybe a marketplace home view in the future
+			page(
+				`/marketplace/product/details/wordpress-seo${
+					selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+				}`
+			);
+		}
+	}, [ dispatch, pluginSlugToBeInstalled ] );
+
+	useEffect( () => {
+		setTimeout(
+			() => {
+				if ( simulatedProgressPercentage < 95 ) {
+					setSimulatedProgressPercentage( ( percentage ) => percentage + 6 );
+				} else if ( simulatedProgressPercentage !== 100 && simulatedProgressPercentage !== 95 ) {
+					setSimulatedProgressPercentage( 95 );
+				}
+			},
+
+			interval
+		);
+	}, [ simulatedProgressPercentage ] );
+
+	useEffect( () => {
+		if ( transferStatus === 'complete' ) {
+			// TODO: Make sure the primary domain is set to the relevant domain before redireting to thank-you page
+			setSimulatedProgressPercentage( 100 );
+			setTimeout( () => {
```

Comment:
`requestAnimationFrame`?

Reply:
Since its a quite slow function (less than 1fps surely) didn't want to use raf. Feel like it is unnecessary complexity for a very slow animation.